src/strace.c

```diff
@@ -2958,7 +2982,10 @@ init(int argc, char *argv[])
 #endif
 
 	/* See if they want to run as another user. */
-	if (username != NULL) {
+	if (username != NULL && cmdline_uid > -1 && cmdline_gid > -1) {
+		run_uid = cmdline_uid;
+		run_gid = cmdline_gid;
```

Comment:
Sorry, it's not clear how --user should interact with --uid/--gid.  Are they expected to work together?

Reply:
Apologies if my answer is not satisfactory. The intent was to run as the given user but without having to look up the user internally. As such the patch kept the requirement for -u, merely supplemented it with --uid/--gid as an optional combo to pass along.

I think unless username is used somewhere inside, the need to pass all three could be replaced with the ability to pass only --uid/--gid.

Reply:
I reworked this a little while adding tests. I'll push the new version soon.

Reply:
Sorry, I don't quite follow why --uid and --gid require --user.  According to the patch, --uid and --gid override whatever is specified by --user, so if that was the intent, then they are mutually exclusive.

Alternatively, one could suggest not to introduce new options but extend --user instead: following the chown(1) example, --user=OWNER:GROUP could be given the meaning of --uid=OWNER --gid=GROUP.

Reply:
Yes, I think this is something that could reduce the number of options. A when --user is OWNER:GROUP and both are numeric, lookup is bypassed. I can update the patch towards that goal.